cli/pkg/local/app.go

```diff
@@ -624,6 +632,49 @@ func initLogger(isVerbose bool, logFormat LogFormat) (logger *zap.Logger, cleanu
 	}
 }
 
+func emitSourceTelemetry(ctx context.Context, opts *AppOptions, rt *runtime.Runtime, inst *drivers.Instance) error {
```

Comment:
Would suggest renaming it to reflect that it emits a start event (the choice of start attributes may increase in the future). Also, these are pretty stateful objects to pass into a util function. Would suggest making it a member function:

```
func (a *App) emitStartEvent(ctx context.Context) error
```